src/sagas/pushNotification.js

```diff
@@ -445,6 +469,77 @@ export function* dismissOptInQuestion() {
   yield STORE.setItem(pushNotificationKey.optInDismissed, true);
 }
 
+/**
+ * This function retrieves the tx details from the wallet history.
+ * @param {Object} wallet the current wallet
+ * @param {string} txId the tx id
+ * @returns {Promise<{
+ *  tx: { txId: string, timestamp: number, voided: boolean },
+ *  tokens: { uid: string, name: string, symbol: string, balance: number, isRegistered: boolean }[]
+ * }>} the tx details
+ * @example
+ * {
+ *   tx: {
+ *     txId: '000021e7addbb94a8e43d7f1237d556d47efc4d34800c5923ed3a75bf5a2886e',
+ *     timestamp: 1673039453,
+ *     voided: false,
+ *   },
+ *   tokens: [
+ *     {
+ *       uid: '00',
+ *       name: 'Hathor',
+ *       symbol: 'HTR',
+ *       balance: 500,
+ *       isRegistered: true,
+ *     }
+ *   ],
+ */
+export const getTxDetails = async (wallet, txId) => {
+  const txTokensBalance = await wallet.getTxById(txId);
+  const [tx] = txTokensBalance;
+  const txDetails = {
+    tx: {
+      txId: tx.txId,
+      timestamp: tx.timestamp,
+      voided: tx.voided,
+    },
+    tokens: txTokensBalance.map((each) => ({
+      uid: each.tokenId,
+      name: each.tokenName,
+      symbol: each.tokenSymbol,
+      balance: each.balance,
+      isRegistered: !!tokens.tokenExists(each.tokenId),
+    })),
+  };
+  return txDetails;
+};
+
+export function setInitialNotificationData(notification) {
+  STORE.setItem(pushNotificationKey.notificationData, notification.data);
+}
+
+/**
+ * This function is responsible for checking if the app was opened by a push notification
+ * and if so, it will load the tx and token data to show the tx detail modal.
+ */
+export function* checkOpenPushNotification() {
+  try {
+    const notificationData = STORE.getItem(pushNotificationKey.notificationData);
+    // Check if the app was opened by a push notification on press action
+    if (notificationData) {
+      STORE.removeItem(pushNotificationKey.notificationData);
+      // Wait for the wallet to be loaded
+      yield take(types.START_WALLET_SUCCESS);
+
+      const wallet = yield select((state) => state.wallet);
+      const txDetails = yield call(getTxDetails, wallet, notificationData.txId);
+      yield put(pushLoadTxDetails(txDetails));
+    }
+  } catch (error) {
```

Comment:
Which part of the function do you expect to have an error?

Reply:
The `getTxDetails` call.